index.js

```diff
@@ -106,6 +107,28 @@ async function fastifyView (fastify, opts) {
 
   const renderer = renders[type] ? renders[type] : renders._default
 
+  async function asyncRender (page) {
+    if (!page) {
+      throw new Error('Missing page')
+    }
+
+    let result = await renderer.apply(this, arguments)
+
+    if (minify && !isPathExcludedMinification(this)) {
+      result = await minify(result, globalOptions.htmlMinifierOptions)
+    }
+
+    if (!this.getHeader('Content-Type')) {
+      this.header('Content-Type', 'text/html; charset=' + charset)
+    }
+
+    return result
+  }
+
+  const fakeRequest = {
+    getHeader: () => true
+  }
```

Comment:
Why a fake request?

Reply:
It's what `this` is bound to in the call to `asyncRender` by the old `viewDecorator` on line 140 -- because viewDecorator is does not have `this` set to a request.